python/ray/serve/tests/test_api.py

```diff
@@ -476,6 +478,106 @@ def f():
             f.set_options(max_concurrent_queries=-4)
 
 
+def test_deploy_application(serve_instance):
+    """Test deploy multiple applications"""
+
+    @serve.deployment
+    def f():
+        return "got f"
+
+    @serve.deployment
+    def g():
+        return "got g"
+
+    @serve.deployment(route_prefix="/my_prefix")
+    def h():
+        return "got h"
+
+    @serve.deployment
+    class Model1:
+        def __call__(self):
+            return "got model1"
+
+    app = FastAPI()
+
+    @serve.deployment(route_prefix="/hello")
+    @serve.ingress(app)
+    class MyFastAPIDeployment:
+        @app.get("/")
+        def root(self):
+            return "Hello, world!"
+
+    f_handle = serve.run(f.bind(), name="app_f")
+    assert ray.get(f_handle.remote()) == "got f"
+    assert requests.get("http://127.0.0.1:8000/").text == "got f"
+
+    g_handle = serve.run(g.bind(), name="app_g", route_prefix="/app_g")
+    assert ray.get(g_handle.remote()) == "got g"
+    assert requests.get("http://127.0.0.1:8000/app_g").text == "got g"
+
+    h_handle = serve.run(h.bind(), name="app_h")
+    assert ray.get(h_handle.remote()) == "got h"
+    assert requests.get("http://127.0.0.1:8000/my_prefix").text == "got h"
+
+    graph_handle = serve.run(
+        DAGDriver.bind(Model1.bind()), name="graph", route_prefix="/my_graph"
+    )
+    assert ray.get(graph_handle.predict.remote()) == "got model1"
+    assert requests.get("http://127.0.0.1:8000/my_graph").text == '"got model1"'
+
+    serve.run(MyFastAPIDeployment.bind(), name="FastAPI")
```

Comment:
please add a few short comments describing what specific behavior is being tested for each of these conditions. helps a lot with readability!